models/eos/interfaces/eos_interfaces.yml

```diff
@@ -0,0 +1,83 @@
+generator_version: 1.0
+metadata:
+  vesion: 1.1
+  status:
+  - preview
+  supported_by:
+  - 'Ansible Network'
+  copyright_str: Copyright 2019 Red Hat
+  license: gpl-3.0.txt
+info:
+  network_os: eos
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manages interface attributes of Arista EOS interfaces.'
+  description:
+  - Manages interface attributes of Arista EOS interfaces.
+  authors:
+  - Nathaniel Case (@qalthos)
+  extends_documentation_fragment: eos
+  notes:
+    - Tested against vEOS 4.20
+schema:
+  type: object
+  description: The schema used for the argspec and docstring
+  properties:
+    config:
+      type: array
+      description: The provided configuration
+      items:
+        $ref: '#/definitions/config_dict'
+    state:
+      $ref: '#/definitions/state'
+
+definitions:
+  config_dict:
+    description:
+    - A dictionary of interface options
+    type: object
+    properties:
+      name:
+        description:
+        - Full name of the interface, i.e. GigabitEthernet1
+        type: str
+        required: true
+      description:
+        description:
+        - Interface description
+        type: str
+      enable:
+        description: Administrative state of the interface.
+        type: bool
+        default: true
+      speed:
+        description:
+        - Interface link speed. Applicable for ethernet interface only.
```

Comment:
```suggestion
        - Interface link speed. Applicable for Ethernet interfaces only.
```

Reply:
Some of these resource models include fixed choices for the speed (auto, 10, 100, etc).  Is that applicable here?